### PR TITLE
ARROW-14510: [R] [CI] ensure that docker runs don't use host-built artifacts

### PR DESCRIPTION
**Pull request by jonkeane**

### CODE REVIEW

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-fedora-r-clang-sanitizer test-r-linux-valgrind test-ubuntu-default-docs

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-fedora-r-clang-sanitizer test-r-linux-valgrind test-ubuntu-default-docs

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-ubuntu-18.04-r-sanitizer test-fedora-r-clang-sanitizer test-r-linux-valgrind test-ubuntu-default-docs

**Comment by jonkeane** [comment]
The two sanitizer failures are legit (and not caused by this ticket — they look like they are from https://github.com/apache/arrow/pull/11850)

File: ci/scripts/r_valgrind.sh [hunk 1]
```diff
@@ -44,3 +49,5 @@ fi
 # https://github.com/microsoft/LightGBM/blob/fa6d356555f9ef888acf5f5e259dca958ca24f6d/.ci/test_r_package_valgrind.sh#L20-L85
 
 popd
+
+popd
```
Comment (thisisnic) [inline ci/scripts/r_valgrind.sh:53]: Is this a typo or is it needed again here?

**Review by thisisnic — APPROVED** [review]
One comment, otherwise LGTM

File: ci/scripts/r_valgrind.sh (hunk 1 above)
Comment (rok) [reply ci/scripts/r_valgrind.sh:53]: Are popds even needed here? My assumption is shell scripts don't influence the state after completing.

**Review by rok — COMMENTED** [review]
LGTM!

Out of curiosity - wouldn't `${R_BIN} CMD INSTALL --clean ${INSTALL_ARGS}` [as suggested here](https://stackoverflow.com/questions/19741160/r-cmd-install-clean-libs-before-a-new-installation/40023267) also solve this problem? (Sorry for my ignorance if this is a completely off mark.)

File: ci/scripts/r_valgrind.sh (hunk 1 above)
Comment (jonkeane) [reply ci/scripts/r_valgrind.sh:53]: I added another `pushd` up above, so I added this down here to match the flow — nut you're right that this shouldn't change the state or anything so is probably not needed.

**Comment by rok** [comment]
> Re: `${R_BIN} CMD INSTALL --clean ${INSTALL_ARGS}` That _should_ solve the problem (though reading [WRE](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Configure-and-cleanup) it was't 100% clear if the `./cleanup` file needed to be up to date with all the files that needed to be removed (in which case it could still fail if it's not up to date for some reason) or if `--clean` would do that on it's own). But more importantly: that would remove the built artifacts from the directory which might force a re-compile elsewhere. This fix intentionally tried to _not_ impact the shared directory between docker and the host so that if the developer relies on keeping the artifacts around between builds, those will not be overwritten or deleted.

Got it! That makes sense, thanks for the explanation :).

**Pull request closed without merging**